local-cli/generator/index.js

```diff
@@ -82,5 +82,13 @@ module.exports = yeoman.generators.NamedBase.extend({
         {name: this.name}
       );
     }
+  },
+
+  install: function() {
+    if (this.options.upgrade) {
+      return;
+    }
+
+    this.npmInstall('react', { '--save': true });
```

Comment:
very good question, might be worth installing the version we know matches react-native's one, we can get it from RN's package.json's peerDependencies.